### DISCUSSION TITLE
suggestion: use share key for api authentication

### DISCUSSION THREAD

**Discussion by bo0tzz** [alangrainger/immich-public-proxy#3] [Feature Requests]
The last part of a share link (`/share/<key>`) can be used for API auth by appending it to the url (`immich.local/api/foo?key=j3lij2w34892etc`). It's probably not a straight swap (I don't know what endpoints you use right now, and the share key is limited) but in theory this should let you remove the need for an API key entirely, reducing the attack surface even more.

**Comment by alangrainger** [comment]
I think you're thinking of the endpoints to get asset info, like [`viewAsset`](https://immich.app/docs/api/view-asset). But to find out what assets are included in a particular share link, I need to use [`getAllSharedLinks`](https://immich.app/docs/api/get-all-shared-links), which requires an API key to access. 

But if there's another way to find out what assets are associated with a given share link (share `key`), I'd love to know, as using `getAllSharedLinks` seems really inefficient since I have to request **all** links and then iterate them.

**Reply by bo0tzz** [reply]
I'm not sure why it's called this, but `/api/shared-links/me?key=asdf` returns the share link details.

**Reply by alangrainger** [reply]
You LEGEND!!! I have no idea why it's called that either, but that is **exactly** what I was looking for!

I will refactor to use that and remove API key completely. Thank you!! :heart:

**Comment by alangrainger** [comment]
API key has now been removed - thank you very much @bo0tzz 

https://github.com/alangrainger/immich-public-proxy/commit/0304b0b787d9e21766f6112d0f978dd455f1b411

**Reply by bo0tzz** [reply]
Awesome! :D

**Comment by ovizii** [comment]
**You might want to align the docs, I quote:**

> It does not require an API key which reduces the attack surface even further.

> To view a shared album in Immich, you need access to the /api/ path. If you're sharing a gallery with the public, you need to make that path public.

**And then further down:**

> Immich Public Proxy takes that key and makes an API call to your Immich instance over your local network, to ask what photos or videos are shared in that share URL.
> 
> If it is a valid share URL, the proxy fetches just those assets via local API and returns them to the visitor as an individual image or gallery.

**Reply by alangrainger** [reply]
The share link random string is called a "key" in the docs. It's not an API key.

edit: I just read my docs and I mention that in the previous paragraph.

**Reply by ovizii** [reply]
I might be having a blonde moment, but it still sounds confusing when you say `it does not require an API key` and then later on you say `it makes an API call`. 

Anyway, it doesn't matter to me, I just thought I'd give you feedback. 

###edit###
Now I get it. You mentioned a public API, which apparently does not require an API key :-) sorry for the confusion.
